lxd/api_internal_recover.go

```diff
@@ -574,12 +574,23 @@ func internalRecoverImportInstanceSnapshot(s *state.State, pool storagePools.Poo
 		return nil, err
 	}
 
+	snapshotExpiry := snap.Config["snapshots.expiry"]
```

Comment:
Do we need this check?

Reply:
`lxd restore` looks for the expiry date in the volume_snapshots section as shown here: https://github.com/canonical/lxd/issues/13464#issuecomment-2307782519. The check I've added imports the instance snapshot with the correct expiry date, but doesn't address the underlying issue of volume_snapshots having zeroed expiry dates in the `backup.yaml` and storage_volumes_snapshots table.  I'm going to look further into this and introduce a better fix.

Reply:
thanks.

>  but doesn't address the underlying issue of volume_snapshots having zeroed expiry dates in the backup.yaml and storage_volumes_snapshots table. I'm going to look further into this and introduce a better fix.

I'm not sure that is an issue TBH, because the associated storage volume for an instance snapshot is always deleted when the instance snapshot is removed.